source/common/network/happy_eyeballs_connection_impl.cc

```diff
@@ -379,6 +380,41 @@ void HappyEyeballsConnectionImpl::dumpState(std::ostream& os, int indent_level)
   }
 }
 
+std::vector<Address::InstanceConstSharedPtr>
+HappyEyeballsConnectionImpl::sortAddresses(const std::vector<Address::InstanceConstSharedPtr>& in) {
+  std::vector<Address::InstanceConstSharedPtr> address_list = in;
+  for (size_t current = 1; current < address_list.size(); ++current) {
+    // If the current address has a different family than the previous address then
+    // it is in the correct position.
+    if (address_list[current]->type() != Address::Type::Ip ||
+        address_list[current - 1]->type() != Address::Type::Ip ||
+        address_list[current]->ip()->version() != address_list[current - 1]->ip()->version()) {
+      continue;
+    }
+    // Find the first address with a different address family than the current address.
+    size_t next = current + 1;
+    while (next < address_list.size()) {
+      if (address_list[next]->type() != Address::Type::Ip ||
+          address_list[current]->ip()->version() != address_list[next]->ip()->version()) {
+        break;
+      }
+      ++next;
+    }
+    // There are no more addresses with different families so sorting is finished.
+    if (next >= address_list.size()) {
+      break;
+    }
+
+    // Bubble the address up to the current position.
+    for (size_t i = next; i > current; i--) {
+      using std::swap;
```

Comment:
Since std::swap is only called once here, would it be cleaner to just call std::swap without the `using` clause?

Reply:
Apparently not, as per go/using-std-swap. Who knew?! (Believe it or not, there is a potential behavior difference between the two)

Reply:
Amazing... Thanks for providing the context!